From 8e9a949def242465356b796094a2ae7f2abe2eba Mon Sep 17 00:00:00 2001
From: Bartlomiej Puget <larhard@gmail.com>
Date: Sat, 25 Aug 2018 12:07:01 +0000
Subject: [PATCH] fix smlnj ebuild

---
 dev-lang/smlnj/smlnj-110.82.ebuild | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dev-lang/smlnj/smlnj-110.82.ebuild b/dev-lang/smlnj/smlnj-110.82.ebuild
index 1df1108..1d76766 100644
--- a/dev-lang/smlnj/smlnj-110.82.ebuild
+++ b/dev-lang/smlnj/smlnj-110.82.ebuild
@@ -116,11 +116,11 @@ remove_element_from_array() {
 # smlnj_get_libdir tries to find the 32 bit lib directory.  Or otherwise,
 # just return the system lib directory $(get_libdir).
 smlnj_get_libdir() {
-	local x=$(get_all_libdirs)
+	local x=( $(get_all_libdirs) )
 	# Remove the native lib dir
-	local y=$(remove_element_from_array $(get_libdir) ${x[@]})
+	local y=( $(remove_element_from_array $(get_libdir) ${x[@]}) )
 	# Remove libx32 if it exists
-	local z=$(remove_element_from_array "libx32" ${y[@]})
+	local z=( $(remove_element_from_array "libx32" ${y[@]}) )
 	# However if the system is not multlib, then we still need to install
 	# the 32 bit smlnj executables and libraries somewhere, so I guess we
 	# just have to put them under the system lib directory.  Put the
-- 
2.16.4

